adafruit_led_animation/animation/multicolor_comet.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-FileCopyrightText: 2022 Tim Cocks
+#
+# SPDX-License-Identifier: MIT
+
+"""
+`adafruit_led_animation.animation.multicolor_comet`
+================================================================================
+
+Multi-color Comet animation for CircuitPython helper library for LED animations.
+
+* Author(s): Kattni Rembor, Tim Cocks
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit NeoPixels <https://www.adafruit.com/category/168>`_
+* `Adafruit DotStars <https://www.adafruit.com/category/885>`_
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://circuitpython.org/downloads
+
+
+"""
+from adafruit_led_animation.animation.comet import Comet
+from adafruit_led_animation.color import BLACK
+
+
+class MulticolorComet(Comet):
+    """
+    A multi-color comet animation.
+
+    :param pixel_object: The initialised LED object.
+    :param float speed: Animation speed in seconds, e.g. ``0.1``.
+    :param colors: Animation colors in a list or tuple of entries in
+                            ``(r, g, b)`` tuple, or ``0x000000`` hex format.
+    :param int tail_length: The length of the comet. Defaults to 25% of the length of the
+                            ``pixel_object``. Automatically compensates for a minimum of 2 and a
+                            maximum of the length of the ``pixel_object``.
+    :param bool reverse: Animates the comet in the reverse order. Defaults to ``False``.
+    :param bool bounce: Comet will bounce back and forth. Defaults to ``True``.
+    :param bool ring: Ring mode.  Defaults to ``False``.
+    :param bool off_pixels: Turn pixels off after the animation passes them. Defaults to ``True``.
+                            Setting to False will result in all pixels not currently in the comet
+                            to remain on and set to a color after the comet passes.
+    """
+
+    # pylint: disable=too-many-arguments,too-many-instance-attributes
+    def __init__(
+        self,
+        pixel_object,
+        speed,
+        colors,
+        tail_length=0,
+        reverse=False,
+        bounce=False,
+        name=None,
+        ring=False,
+        off_pixels=True,
+    ):
```

Comment:
Not a gating comment, but it may be worth using the `*` sentinel from initial release of this to ensure all of the `bool` arguments are passed as keyword-only.